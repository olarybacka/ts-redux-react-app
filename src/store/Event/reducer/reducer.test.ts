import { actionCreators, Event, eventReducer, INITIAL_STATE } from ".."

export const testEvent: Event = {
  id: 1,
  name: "Name",
  startDate: expect.any(Date),
  endDate: expect.any(Date),
  description: "description",
  location: "location",
  category: {
    id: 1,
    name: "name",
  },
}

describe("EventReducer", () => {
  it("calls updateEvents action", () => {
    const action = {
      type: actionCreators.updateEvents.type,
      payload: [testEvent],
    }
    expect(eventReducer(INITIAL_STATE, action)).toMatchSnapshot()
  })
  it("calls updateEvent action", () => {
    const action = {
      type: actionCreators.updateEvent.type,
      payload: testEvent,
    }
    expect(eventReducer(INITIAL_STATE, action)).toMatchSnapshot()
  })
})
