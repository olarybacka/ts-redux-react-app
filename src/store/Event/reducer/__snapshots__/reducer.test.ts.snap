// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventReducer calls updateEvent action 1`] = `
Object {
  "count": 0,
  "error": "",
  "event": Object {
    "category": Object {
      "id": 1,
      "name": "name",
    },
    "description": "description",
    "endDate": Any<Date>,
    "id": 1,
    "location": "location",
    "name": "Name",
    "startDate": Any<Date>,
  },
  "events": Array [],
  "next": null,
  "previos": null,
}
`;

exports[`EventReducer calls updateEvents action 1`] = `
Object {
  "count": 0,
  "error": "",
  "event": Object {
    "category": null,
    "description": "",
    "endDate": null,
    "id": NaN,
    "location": "",
    "name": "",
    "startDate": null,
  },
  "events": Array [
    Object {
      "category": Object {
        "id": 1,
        "name": "name",
      },
      "description": "description",
      "endDate": Any<Date>,
      "id": 1,
      "location": "location",
      "name": "Name",
      "startDate": Any<Date>,
    },
  ],
  "next": null,
  "previos": null,
}
`;
